Pricing guide (#702)
liveblocks/liveblocks: docs/pages/guides/liveblocks-pricing-model.mdx

---
meta:

>   title: "Liveblocks 1.0 pricing model"

Reviewer: ```suggestion
  title: "Pricing"
```

Reviewer: I would go with something simple. It looks off in the menu if too long.

  parentTitle: "Guides"
  description: "Explanation of Liveblocks 1.0 pricing model and implementation"
---

With the release of Liveblocks 1.0, we included new pricing plans that are based

> on monthly active users (MAU). This guide will help you understand how to use

Reviewer: ```suggestion
on Monthly Active Users (MAU). This guide will help you understand how to use
```

the new pricing plans by providing guidance on the migration to Liveblocks 1.0
and how to include a `userId` into the `authorize` option if you use it.

> ## What is a Monthly Active User (MAU) vs. a connection? [#mau-vs-connection]

Reviewer: ```suggestion
## What is a Monthly Active User (MAU)? [#what-is-mau]
```

I'd go with something simpler

In versions before 1.0 we used the number of connections to determine pricing
segments. We counted each time a user connected to a Liveblocks room from a

> physical device as a connection. While this approximated the number of users, it

Reviewer: It never approximated the number of users. Dialogue has 500 MAU, and 4M connections. The whole reason why we did MAU is to map the pricing to the number of users

did not account for short-lived connections and was sometimes hard to predict.
We introduced pricing based on Monthly Active Users (MAU) to reflect the value
that Liveblocks offers to our customers.

A monthly active user can be defined as:

> - Each unique `userId` that connects to a Liveblocks room at least once a month

Reviewer: ```suggestion
- A `userId` that connects to a Liveblocks room at least once a month
```

Since we're saying before: "A monthly active user can be defined as:"

  (if using authorization)

> - Each unique cookie that connects to a Liveblocks room at least once a month

Reviewer: ```suggestion
- An anonymous user, identified by a Liveblocks cookie, that connects to a Liveblocks room at least once a month
```

That sounds weird...

Author: I re-worded, let me know if knew works for you

  (if not using authorization)

## How does authorization work with MAU? [#authorization]

In `@liveblocks/node`, we made the field `userId` in the `authorize` option
mandatory. This `userId` is used to track MAU associated with a Liveblocks

> account. Each `userId` will count as 1 MAU. To make sure you begin to track MAU
> as intended, first update `@liveblocks/node`.

Reviewer: ```suggestion
Each `userId` is equivalent to one MAU. To ensure accurate tracking of MAU, start by updating @liveblocks/node.
```

"first update xxx' sounded weird, what do you think of that?

```bash
npm update @liveblocks/node@1.0
```

Then, update your `authorize` option to include the `userId` field.

```ts highlight="16,20"
import { authorize } from "@liveblocks/node";

// Replace this key with your secret key provided at
// https://liveblocks.io/dashboard/projects/{projectId}/apikeys
const secret = "sk_prod_xxxxxxxxxxxxxxxxxxxxxxxx";

export default async function auth(req, res) {
  /**
   * Implement your own security here.
   *
   * It's your responsibility to ensure that the caller of this endpoint
   * is a valid user by validating the cookies or authentication headers
   * and that it has access to the requested room.
   */
  const room = req.body.room;
  const response = await authorize({
    room,
    secret,
    // Corresponds to the UserMeta[id] type defined in liveblocks.config.ts

>     userId: "123",

Reviewer: Why isn't it comment as required like in the example of the 1.0 upgrading guide?

    groupIds: ["456"], // Optional
    userInfo: {
      // Optional, corresponds to the UserMeta[info] type defined in liveblocks.config.ts
      name: "Ada Lovelace",
      color: "red",
    },
  });
  return res.status(response.status).end(response.body);
}
```

<Banner title="Using a secret key with authorization">
  To use the `authorize` option, you need your backend endpoint. Your Liveblocks
  dashboard contains the secret key required to use the `authorize` option.
</Banner>

To learn more about authorization and authentication in your Liveblocks

> application, check out our [Authentication Guide](/docs/guides/authentication).

Reviewer: ```suggestion
application, check out our [Authentication guide](/docs/guides/authentication).
```

why use Camel Case here?

## How is MAU calculated using a public key? [#public-key]

To take advantage of Liveblocks 1.0 pricing without authorization, you must
update your Liveblocks packages to 1.0. We will count each connection as one MAU
if you do not update your packages to 1.0. Once the packages are updated, a
cookie will be set on the client to track when a new user connects to a room.
This cookie will track MAU associated with a Liveblocks account and expire after
30 days.

Inside the `createClient` function, you will use your public key instead of a
private key.

```ts file="liveblocks.config.ts" highlight="4"
import { createClient } from "@liveblocks/client";

const client = createClient({
  publicApiKey: "pk_prod_xxxxxxxxxxxxxxxxxxxxxxxx",
});
```

To illustrate how counting MAU works without a `userId`, consider the following:

If you have one user who connects to a Liveblocks room that does not enforce

> authorization, you will be billed for 1 MAU. That same person can reaccess a

Reviewer: ```suggestion
authorization, you will be billed for one MAU. That same person can reaccess a
```

room the next day, and MAU would not increase. The cookie will expire at the end
of the 30-day billing cycle, and the user will be counted as a new MAU.
Liveblocks will count that anonymous user as 2 MAU if packages are not updated.

> ## Development Use [#development-use]

Reviewer: ```suggestion
## Development use [#development-use]
```

It shoun't be Camel Case, but Sentence case

> While working in development mode, you will not be billed for MAU. However, you

Reviewer: This is too vague, what is "development mode"? We should mention it's localhost users

will still need to include a `userId` in the `authorize` option if you use it.

> We are currently working on updating our dashboard to reflect the number of MAU
> for development use.

Reviewer: ```suggestion
We are currently working on updating our dashboard to identify the MAUs coming from `localhost` against others.
```

## How to migrate to Liveblocks 1.0 [#migration]

> To update to Liveblocks 1.0, you will need to update your Liveblocks packages:

Reviewer: ```suggestion
To update to Liveblocks 1.0, you will need to update your liveblocks packages:
```

In trrhe upgrading guide, you put it in lower case. Shouldn't we keep the same logic here?

Author: yes, I'll keep lowercase and ask for clarification on which is preferred

```bash
npm update @liveblocks/node @liveblocks/client @liveblocks/react
```

You will then pick one of the two methods outlined above for tracking MAU:

> [authorization](#authorization) or [public](#public-key). After completing these

Reviewer: ```suggestion
- [authorization method with secret key](#authorization), or
- [public key](#public-key). 
After completing these
```

"public" felt too vague

steps to take advantage of the new pricing model, you should review the

> [Liveblocks 1.0 Upgrade Guide](/docs/guides/upgrading/1.0) for additional

Reviewer: ```suggestion
[Liveblocks 1.0 upgrading guide](/docs/guides/upgrading/1.0) for additional
```
